### PR TITLE
Add BLIP-2 to eval suite

### PR DESCRIPTION
**Pull request by anas-awadalla**
Added a BLIP-2 model class to our test bed. This should help to validate that our implementation is correct.

I am able to match the OKVQA and COCO numbers for [BLIP-2 T5-XL](https://huggingface.co/Salesforce/blip2-flan-t5-xl) and [BLIP-2 T5-XL COCO](https://huggingface.co/Salesforce/blip2-flan-t5-xl-coco). I have a VQAv2 evaluation run going that should finish in a bit.

One thing I have noticed is that the new InstructBLIP paper includes Flickr30k numbers for the older models. For some reason our cider scores on Flickr are lower than reported (64 vs 76). I will read the InstructBLIP paper more closely and debug if needed.

Edit: VQAv2 also checks out.

### CODE REVIEW

**Comment by ludwigschmidt** [comment]
Great! Validating the eval suite is always super important.

File: open_flamingo/eval/models/blip.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            batch_images = torch.cat(
+                [
+                    batch_images,
+                    self.processor.image_processor(example, return_tensors="pt")[
+                        "pixel_values"
+                    ],
+                ]
+                if batch_images is not None
+                else [
+                    self.processor.image_processor(example, return_tensors="pt")[
```
[the remark this answers is not in the document]
Comment (anas-awadalla) [reply open_flamingo/eval/models/blip.py:63]: this is what the liner wants lol

File: open_flamingo/eval/evaluate.py [hunk 2]
```diff
@@ -216,6 +218,9 @@ def main():
     module = importlib.import_module(f"open_flamingo.eval.models.{args.model}")
```
[the remark this answers is not in the document]
Comment (anas-awadalla) [reply open_flamingo/eval/evaluate.py:218]: You would never need to import more than a single model in a run if that is what you mean. Currently we only have 2 models and I am not sure if we have plans on adding more (although I hope we do).

File: open_flamingo/eval/models/blip.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            max_length=2000,
+        )
+        input_ids = encodings["input_ids"]
+        attention_mask = encodings["attention_mask"]
+
+        with torch.inference_mode():
+            outputs = self.model.generate(
+                self._prepare_images(batch_images).to(self.device),
+                input_ids.to(self.device),
+                attention_mask=attention_mask.to(self.device),
```
[the remark this answers is not in the document]
Comment (anas-awadalla) [reply open_flamingo/eval/models/blip.py:96]: Where is the weird blank line specifically?

File: open_flamingo/eval/evaluate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpgard) [reply open_flamingo/eval/evaluate.py:218]: ah ok I see why this makes sense, nvm 🙃

**Pull request merged**